src/meteors/attr/attributes.py

```diff
@@ -0,0 +1,553 @@
+    if attributes.shape != hsi.image.shape:
+        raise ValueError("Attributes must have the same shape as the hsi")
+
+
+def align_band_names_with_mask(band_names: dict[str, int], band_mask: torch.Tensor) -> dict[str, int]:
+    """Aligns the band names dictionary with the unique values in the band mask.
+
+    This function ensures that the band_names dictionary correctly represents all unique
+    values in the band_mask. It adds a 'not_included' category if necessary and validates
+    that all mask values are accounted for in the band names.
+
+    Args:
+        band_names (dict[str, int]): A dictionary mapping band names to their corresponding
+                                     integer values in the mask.
+        band_mask (torch.Tensor): A tensor representing the band mask, where each unique
+                                  integer corresponds to a different band or category.
+
+    Returns:
+        dict[str, int]: The updated band_names dictionary, potentially including a
+                        'not_included' category if 0 is present in the mask but not in
+                        the original band_names.
+
+    Raises:
+        ValueError: If the set of values in band_names doesn't match the unique values
+                    in the band_mask after accounting for the 'not_included' category.
+
+    Warns:
+        UserWarning: If a 'not_included' category (0) is added to the band_names.
+
+    Notes:
+        - The function assumes that 0 in the mask represents 'not_included' areas if
+          not explicitly defined in the input band_names.
+        - All unique values in the mask must be present in the band_names dictionary
+          after the alignment process.
+    """
+    unique_mask_values = set(band_mask.unique().int().tolist())
+    band_name_values = set(band_names.values())
+
+    # Check if 0 is in the mask but not in band_names
+    if 0 in unique_mask_values and 0 not in band_name_values:
+        warnings.warn(
+            "Band mask contains `0` values which are not covered by the provided band names. "
+            "Adding 'not_included' to band names."
+        )
+        if "not_included" in band_names:
+            raise ValueError(
+                "Band names should not contain 'not_included' if 0 is present in the mask and not in band names"
+            )
+        band_names["not_included"] = 0
+        band_name_values.add(0)
+
+    # Validate that all mask values are in band_names
+    if unique_mask_values != band_name_values:
+        raise ValueError("Band names should have all unique values in mask")
+
+    return band_names
+
+
+def validate_attribution_method(value: str | None) -> str | None:
+    if value is None:
+        return value
+    value = value.title()
+    if value not in AVAILABLE_ATTRIBUTION_METHODS:
+        logger.warning(
+            "Unknown attribution method: {value}. The core implemented methods are {AVAILABLE_ATTRIBUTION_METHODS}"
+        )
+    return value
```

Comment:
I used to raise a warning, but in case someone writes their own method, then one needs to modify the package code. If it is a warning - there is no problem with that